Library/Homebrew/extend/os/linux/diagnostic.rb

```diff
@@ -0,0 +1,65 @@
+require "tempfile"
+
+module Homebrew
+  module Diagnostic
+    class Checks
+      alias generic_check_tmpdir_sticky_bit check_tmpdir_sticky_bit
+
+      def check_tmpdir_sticky_bit
+        message = generic_check_tmpdir_sticky_bit
+        return if message.nil?
+        message + <<~EOS
+          If you don't have administrative privileges on this machine,
+          create a directory and set the HOMEBREW_TEMP environment variable,
+          for example:
+            export HOMEBREW_TEMP=~/tmp
+            mkdir -p "$HOMEBREW_TEMP"
+            chmod 1755 "$HOMEBREW_TEMP"
+            echo "export HOMEBREW_TEMP=\"$HOMEBREW_TEMP\"" >> #{shell_profile}
```

Comment:
Are no `Utils::Shell` methods useful here?

Reply:
Yes, both `export_value` looks useful.
A new function `export_value_in_profile` similar to `prepend_path_in_profile` would be quite useful.

Reply:
Will do. Note, however, that `prepend_path_in_profile` should, in fact, be `append_path_in_profile`.

Reply:
I take it back. The path it prepended :) No one cares that this is done at the end of a `profile` file.

Reply:
@maxim-belkin We don't normally tell people to `export` as well as the shell changes; the assumption are people will open a new shell after the change.

Reply:
I'm tempted to say remove the use of the `HOMEBREW_TEMP` bash variable here (while still setting it in the profile) altogether.

Reply:
> We don't normally tell people to export as well as the shell changes; the assumption are people will open a new shell after the change.

That instruction would need to be given explicitly, if so.

Reply:
> That instruction to close the current shell and open a new one ought to be given explicitly, in that case. I'd prefer to export.

It's not elsewhere and I think consistency is more important.

Reply:
I defer.

Reply:
@sjackman 🙇

Reply:
`install` seems to be present on each of the systems (Docker images) that I quickly tested.